tests/XRootD/http.sh

```diff
@@ -189,4 +189,62 @@ function test_http() {
   receivedHeader=$(grep -i 'Test:' "$outputFilePath")
   assert_eq "1" "$(echo "$receivedHeader" | wc -l | sed 's/^ *//')" "Incorrect number of 'Test' header values"
   assert_eq "$expectedHeader" "$receivedHeader" "HEAD is missing statically-defined Test header"
+
+  set -ex
+
+  run_and_assert_http_and_error_code() {
+    local expected_http_code="$1"
+    local expected_error_code="$2"
+    shift 2
+
+    local body_file
+    body_file=$(mktemp)
+    local http_code
+
+    # Run the curl command, capture HTTP code and body
+    http_code=$(curl -s -L -w "%{http_code}" -o "$body_file" "$@")
+    local body
+    body=$(< "$body_file")
+    rm -f "$body_file"
+
+    # Print body to stderr for debug visibility
+    echo "$body" >&2
```

Comment:
Delay printing out the body until an error has occurred.  That'll keep the tests less chatty for successes.